apps/server/src/modules/news/uc/news.uc.ts

```diff
@@ -76,7 +76,9 @@ export class NewsUc {
 		// by default show latest news first
 		if (options.order == null) options.order = { displayAt: SortOrder.desc };
 
-		const [newsList, newsCount] = await this.newsRepo.findAll(targets, unpublished, options);
+		const [newsList, newsCount] = unpublished
+			? await this.newsRepo.findAllUnpublished(targets, userId, options)
+			: await this.newsRepo.findAllPublished(targets, options);
```

Comment:
if unpublished === true, and user does not have NEWS_EDIT permission, then it should not call findAllUnpublished